lib/liquid/tags/snippet.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Liquid
+  # @liquid_type tag
+  # @liquid_category theme
+  # @liquid_name snippet
+  # @liquid_summary
+  #   Creates a new inline snippet.
+  # @liquid_description
+  #   You can create inline snippets to make your Liquid code more modular.
+  # @liquid_syntax
+  #   {% snippet input %}
+  #     value
+  #   {% endsnippet %}
+
+  class Snippet < Block
+    SYNTAX = /(#{VariableSignature}+)/o
+
+    def initialize(tag_name, markup, options)
+      super
+      p = @parse_context.new_parser(markup)
+      if p.look(:id)
+        @to = p.consume(:id)
+      else
+        raise SyntaxError, options[:locale].t("errors.syntax.snippet")
+      end
+    end
+
+    def render_to_output_buffer(context, output)
+      snippet_drop = SnippetDrop.new(@body)
+      context.scopes.last[@to] = snippet_drop
```

Comment:
I believe we could have something like a [`with_snippet`](https://github.com/Shopify/liquid/blob/1c1e7119065b0fa915149b9e2bdc303490a0dfff/lib/liquid/tags/capture.rb#L35) method here. This way, we make sure the assignments for inline snippets will respect the [resource limits](https://github.com/Shopify/liquid/blob/1c1e7119065b0fa915149b9e2bdc303490a0dfff/lib/liquid/resource_limits.rb#L11) :)